server/src/modules/campaign-monitor/campaign-monitor.service.ts

```diff
@@ -1690,6 +1767,32 @@ export default class CampaignMonitorService {
 		}
 	}
 
-	// TODO: validate if CM is up and running
-	// TODO: validate if LDAP is up and running
+	static async initialMigrationPrepareSubscribers(): Promise<void> {
+		const listIds = [
+			process.env.CAMPAIGN_MONITOR_OPTIN_LIST_05 as string,
+			process.env.CAMPAIGN_MONITOR_NEWSLETTER_LIST_ID as string,
+			process.env.CAMPAIGN_MONITOR_WORKSHOP_LIST_ID as string,
+			process.env.CAMPAIGN_MONITOR_AMBASSADOR_LIST_ID as string,
+		];
+
+		return Promise.all([this.getAllSubscribersFromCMLists(listIds), this.getAllLdapProfiles()])
+			.then(() => this.createMigrationList())
+			.catch((err) => {
+				throw new CustomError(
+					`Failed to run the initial migration (part 1: prepare subscribers)`,
+					err
+				);
+			});
+	}
+
+	static async initialMigrationPushCm(): Promise<void> {
+		console.info('List 5 migration - 2: Starting list 5 migration');
+		return this.executeAndValidateList5Migration();
```

Comment:
Do not use the return keyword for functions that do not return anything.

```suggestion
		await this.executeAndValidateList5Migration();
```